llms/mlx_lm/lora.py

```diff
@@ -81,6 +82,12 @@ def build_parser():
         type=str,
         help="Directory with {train, valid, test}.jsonl files",
     )
+    parser.add_argument(
+        "--train-full",
+        action="store_true",
+        help="Train the full model without LoRA layers.",
+        default=False,
+    )
```

Comment:
It might be better to have a `--fine-tune-type` flag which can be any of `lora`, `dora`, or `full` (and default to `lora) and remove `train-full` and `use-dora` as flags. That will be more extensible to other methods as well should we ever want to do so. Wdyt?

Reply:
yes that sounds best, I'll update the code 👍

Reply:
Thanks!

Reply:
Hey, I'm back, it looks like its working for me, removed `use-dora` and added `--fine-tune-type`, changed `--adapter-path` to `--adapter-model-path `and `--resume-adapter-file` to `--resume-adapter-model-file` for both training phases. Also updated the `save_every` part to `save adapter` or `model` automatically. Let me know if this need some changing.

Reply:
This is what i tried:

```sh
python -m mlx_lm.lora \
    --model Xenova/tiny-random-GemmaForCausalLM \
    --fine-tune-type dora \
    --train \
    --data /Users/gokdenizgulmez/Library/Mobile\ Documents/com\~apple\~CloudDocs/Datastes/data_tyni \
    --iters 2 \
    --batch-size 1 \
    --lora-layers 2 \
    --val-batches 1 \
    --adapter-model-path /Users/gokdenizgulmez/Desktop/tinyGemma-pretrain



python -m mlx_lm.lora \
    --model Xenova/tiny-random-GemmaForCausalLM \
    --train \
    --fine-tune-type full \
    --data /Users/gokdenizgulmez/Library/Mobile\ Documents/com\~apple\~CloudDocs/Datastes/data_tyni \
    --iters 2 \
    --batch-size 1 \
    --val-batches 1 \
    --adapter-model-path /Users/gokdenizgulmez/Desktop/tinyGemma-pretrain
```